### PR TITLE
Allow aggregations to work with multi-tab.

### LINKED ISSUE #7198: Failed to get aggregate result because the client is offline
* Browser version: **Chrome 111.0.5563.146**
  * Firebase SDK version: **firebase 9.15**
  * Firebase Product: **firestore database**
  *  React Application compiles via Vite.

I wanted to start using the new aggregate queries, to be more efficient in my reads.

My project has collections of `projects` and `tasks`

I wanted to find the totals of tasks per project with the following query, through a `forEach()`

```javascript
projectIds.forEach((projectId: string) => {
  const allTasksQuery = query(
	  firebase.documents(),
	  where('taskEmails', 'array-contains', encodedEmail),
	  where('project.uid', '==', projectId),
	  where('isLatestVersion', '==', true)
  );
  const promise = getCountFromServer(allTasksQuery)
	  .then((result) => {
		  const projectCount = result.data().count;
		  totalTasks[projectId] = projectCount;
	  })
	  .catch((err) => console.error('Error getting task Totals', err));
  allPromises.push(promise);
  });
```

NOTE: `firebase.documents()` is just a wrapper for this ref:
```javascript
	const documents = () => collection(firestore, 'documents');
```
This works roughly 50% of the time, especially when coming from a page that has already recieved data back from firestore in a more convention `getDoc` or `getDocs` query.  If I hard reload on the page that executes this query, as well as in other circumstances I receive the following error:

`FirebaseError: Failed to get aggregate result because the client is offline.`
image

Any help or clarification would be greatly appreciated.  Thank-you.

### PR DESCRIPTION
**Pull request by ehsannas**
Fixes #7198.

A proper long-term solution for multi-tab should be implemented.

### CODE REVIEW

File: .changeset/three-months-lie.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@firebase/firestore": patch
+---
+
+Fix a bug that prevented aggregations from being run when muli-tab was enabled.
```
Comment (dconeybe) [inline .changeset/three-months-lie.md:5]: nit: I'd just recommend saying "sometimes" in this change log entry. The way it is worded IMO incorrectly implies that count queries didn't work at all in multi-tab mode.

Comment (dconeybe) [inline .changeset/three-months-lie.md:5]: nit: "multi-tab" -> "multi-tab persistence"

**Review by dconeybe — APPROVED** [review]
Two minor comments about the change log entry.

File: .changeset/three-months-lie.md (hunk 1 above)
Comment (ehsannas) [reply .changeset/three-months-lie.md]: Done

Comment (ehsannas) [reply .changeset/three-months-lie.md]: Done

**Pull request merged**